sdk/python/kfp/cli/component_test.py

```diff
@@ -608,7 +605,7 @@ def test_dockerfile_can_contain_custom_kfp_package(self):
         file_start = textwrap.dedent('''\
                 # Generated by KFP.
 
-                FROM python:3.9
+                FROM python:3.12
```

Comment:
based on your comment, it seems like it works for 3.9, if so maybe it's best if we keep this consistent with the rest of the tests and use 3.9

Reply:
Unfortunately it doesn't. I'd executed a subprocess previously in my virtual environment and then ran this test which is why it went through. 

I replicated this on a fresh venv and [the missing wheel file error](https://github.com/kubeflow/pipelines/issues/11038) still comes up.